plotly_resampler/figure_resampler/figure_resampler_interface.py

```diff
@@ -970,12 +1026,12 @@ def add_trace(
                 )
             else:
                 self._print(f"[i] NOT resampling {trace['name']} - len={n_samples}")
-                for k in dc._fields:
-                    setattr(trace, k, getattr(dc, k))
+                trace._process_kwargs(**{k: getattr(dc, k) for k in dc._fields})
                 return super(AbstractFigureAggregator, self).add_traces(
                     [trace], **self._add_trace_to_add_traces_kwargs(trace_kwargs)
                 )
-        return super(AbstractFigureAggregator, self).add_traces(
+
+        return super(self._figure_class, self).add_traces(
```

Comment:
Why do we use here`self.figure_class`?

Reply:
Is this to reduce the double call of `add_traces`?

Reply:
Could be! I believe you investigated this before..